README.md
test_repository
===============

test repository for scicoder workshop


history.txt
   37  oplot, lambda, mspect[*,incy,inct,inca,inp,inH,inssa,i], color=101+i&$
   38  endfor
   39  idl
   40  date
   41  idl
   42  source configenv
   43  idl
   44  cd analysis/work/mm/kdk/
   45  scp -r sluszcz@beast.berkeley.edu:/indirect/mozzy1/kdekleer/carma/Io2013a_C2b/*.csh .
   46  amacs Io_script.csh 
   47  ssh -Y sluszcz@beast.berkeley.edu
   48  source configenv
   49  idl
   50  source configenv
   51  idl
   52  cd /usr/local/lib/
   53  ls
   54  amacs .profile
   55  cd analysis/work/mm/kdk/
   56  ls
   57  amacs Io_script_statia.csh &
   58  ./Io_script_statia.csh selfcalcal
   59  retall
   60  source /Users/statia/miriad/source miriad_start.sh
   61  source /Users/statia/miriad/miriad_start.sh
   62  ./Io_script_statia.csh selfcalcal
   63  ./Io_script_statia.csh startup
   64  locate libiconf.2.dylib
   65  locate libiconf.3.dylib
   66  locate libgfortran.3.dylib
   67  otool -L opt/local/lib/gcc45/libgfortran.3.dylib
   68  locate libgfortran
   69  cd /usr
   70  locate libgfortran
   71  cd
   72  cd Downloads/
   73  ls -ltr
   74  sudo tar -xvf gcc-mlion.tar -C /
   75  cd
   76  cd /usr/local/lib/
   77  ls
   78  cd
   79  ftp://ftp.astro.umd.edu/progs/carma/miriad_mac10.8_gfortran.tar.gz | tar zxf -
   80  curl ftp://ftp.astro.umd.edu/progs/carma/miriad_mac10.8_gfortran.tar.gz | tar zxf -
   81  ls
   82  ls -ltr
   83  cd miria
   84  cd miriad
   85  wget ftp://ftp.astro.umd.edu/progs/carma/miriad_mac10.8_gfortran.tar.gz
   86  curl ftp://ftp.astro.umd.edu/progs/carma/miriad_mac10.8_gfortran.tar.gz | tar zxf -
   87  curl ftp://ftp.astro.umd.edu/progs/carma/miriad_mac10.8.tar.gz | tar zxf -
   88  cd miriad
   89  install/make_miriad_starts
   90  source miriad_start.csh
   91  source miriad_start.sh
   92  cd $MIR
   93  ls
   94  pwd
   95  more VERSION
   96    install/test_a_new_miriad_cvs rootdir=. reuse=1 auto=1
   97  install/install.miriad
   98  ls
   99  cd install
  100  ls
  101  amacs install.miriad
  102  ls
  103  cd ..
  104  ls
  105  more VERSIONA
  106  more VERSIONS
  107  cd install
  108  install/install.miriad
  109  cd ..
  110  install/install.miriad
  111  install/install.miriad
  112  cd ..
  113  curl ftp://ftp.astro.umd.edu/progs/carma/miriad.tar.gz | tar zxf -
  114  cd miriad
  115    cvschroot :pserver:$USER@cvs.astro.umd.edu:/home/cvsroot
  116  cvs login
  117  cvs login
  118   cvs update -d
  119  amacs install/install.miriad
  120  install/install.miriad
  121  cd miriad/
  122  install/install.miriad
  123  install/install.miriad
  124  cd
  125  curl ftp://ftp.astro.umd.edu/progs/carma/miriad_mac10.8.tar.gz | tar zxf - 
  126  curl ftp://ftp.astro.umd.edu/progs/carma/miriad_mac10.8.tar.gz | tar zxf - 
  127  cd miriad
  128  install/make_miriad_starts
  129  source miriad_start.sh
  130  date
  131  date
  132  date
  133  idl
  134  exit
  135  cd /Applications/ITT/license/
  136  ls
  137  scp sluszcz@beast.berkeley.edu:/indirect/apps3/rsi/idl_8.1/license/license.data.saturn .
  138  ls
  139  scp sluszcz@beast.berkeley.edu:/indirect/apps3/rsi/idl_8.1/license/license.dat.saturn .
  140  ls
  141  more readme.txt 
  142  pwd
  143  ls
  144  mv license.dat.lupus license.dat.lupus.broke
  145  ls
  146  ls -ltr
  147  cnmod +rwx license.dat.saturn 
  148  ls -ltr
  149  chmod +rwx license.dat.saturn 
  150  ls
  151  ls -ltr
  152  chmod +777 license.dat.saturn 
  153  chmod +755 license.dat.saturn 
  154  chmod +744 license.dat.saturn 
  155  chmod 744 license.dat.saturn 
  156  ls -ltr
  157  chmod 777 license.dat.saturn 
  158  ls -ltr
  159  mv license.dat.lupus.broke license.dat.lupus
  160  pwd
  161  more license.dat.saturn 
  162  idl
  163  idl
  164  idl
  165  idl
  166  echo $LM_LICENSE_FILE
  167  cd analysis/work/nir/multiwavel_2003/data/
  168  idl
  169  idl
  170  cd /Applications/ITT/license/
  171  ls
  172  amacs license.dat.lupus 
  173  amacs license.dat.saturn 
  174  source .idlenv 
  175  amacs .idlenv 
  176  cd /Applications/ITT/license/
  177  ls
  178  more license.dat.lupus 
  179  ssh -Y  sluszcz@beast.berkeley.edu
  180  date
  181  cd analysis/work/nir/multiwavel_2003/
  182  cd data/
  183  curl  http://astro.berkeley.edu/~imke/statia/reg_files/*.gz
  184  ls
  185  curl  "http://astro.berkeley.edu/~imke/statia/reg_files/*.gz"
  186  pwd
  187  idl
  188  idl
  189  idl
  190  export Applications/ITT/license/license.dat.saturn
  191  export LM_LICENSE_FILE='/Applications/ITT/license/license.dat.saturn'
  192  idl
  193  source .idlenv
  194  source ~/.idlenv
  195  idl
  196  echo $LM_LICENSE_FILE
  197  amacs /Applications/ITT/license/license.dat.saturn
  198  source ~.idlenv
  199  source ~/.idlenv
  200  idl
  201  source ~/.idlenv
  202  idl
  203  source configenv
  204  id
  205  id
  206  idl
  207  source configenv
  208  idl
  209  source configenv
  210  id
  211  idl
  212  source configenv
  213  idl
  214  ssh -Y sluszcz@beast.berkeley.edu
  215  cd analysis/work/mm/
  216  ls
  217  cd data/
  218  ls
  219  cd ..
  220  cd map/
  221  ls
  222  cd final/
  223  ls
  224  cd uv/
  225  ls
  226  scp -r sluszcz@beast.berkeley.edu:/indirect/mozzy11/sluszcz/workCARMA/final/uv/daily/D_11mar2009.uv .
  227  ls
  228  uvlist options=spectra vis=D_11mar2009.uv
  229  ls
  230  rm -r D_11mar2009.uv/
  231  ls
  232  csh
  233  source .profile
  234  source ~/.profile
  235  .csh
  236  csh
  237  amacs .idlenv
  238  amacs .profile
  239  sudo port sync
  240  sudo port -vunk upgrade miriad
  241  date
  242  cd analysis/work/nir/multiwavel_2003/
  243  idl
  244  date
  245  cd analysis/work/nir/neptune_repolocal/
  246  source configenv
  247  idl
  248  idl
  249  cd analysis/work/nir/neptune_repolocal/
  250  source configenv
  251  idl
  252  amacs .idlenv
  253  amacs idlstartup.pro
  254  ls
  255  amacs .cshrc
  256  amacs .profile
  257  amacs .idlenv
  258  idl
  259  source configenv
  260  idl
  261  cd analysis/work/nir/src/
  262  hg pull
  263  date

spect2.py
from astropy.io import fits
import numpy as np
import copy
import pylab
import math


class datafile:
	#def __init__(self, path, variableList=["LOGG_ADOP", "FEH_ADOP", "TEFF_ADOP", "SPECTYPE_HAMMER"]):
	def __init__(self, path, variableList=["LOGG_ADOP", "FEH_ADOP", "TEFF_ADOP", "SPECTYPE_HAMMER"]):
		self.path = path
		datafile = fits.open(path)
		#self.dataDict = {}
		#for variable in variableList:
		#	self.dataDict[variable] = np.array(datafile[1].data.field(variable))
		self.logg = np.array(datafile[1].data.field("LOGG_ADOP"))
		self.feh = np.array(datafile[1].data.field("FEH_ADOP"))
		self.teff = np.array(datafile[1].data.field("TEFF_ADOP"))
		self.spectypehammer = np.array(datafile[1].data.field("SPECTYPE_HAMMER"))
		self.clean()
		
	def clean(self):
		self.logg, self.feh, self.teff, self.spectypehammer = cleanData([self.logg, self.feh, self.teff, self.spectypehammer])
		#cleanedDataList = cleanData(self.dataDict.values())
		
	def uniqueSpectralTypes(self):
		return list(set(self.spectypehammer))
		
	def plotHist(self):
		fig,axes = pylab.subplots(1, 3)
		axes[0].hist(self.feh, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled', bins=20)
		axes[1].hist(self.feh, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled')
		axes[2].hist(self.feh, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled')
		#axes[1].hist(self.PUTRADIALVELOCITYARRAYHERE, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled')
		#axes[2].hist(self.PUTHELIODISTARRAYHERE, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled')
		pylab.show()
	

def cleanData(arrayList, badValue=-9999):
	goodIndices = []
	for i in xrange(len(arrayList[0])):
		foundBadValue = False
		for array in arrayList:
			assert len(arrayList[0]) == len(array)
			if array[i] == badValue: foundBadValue = True
		if not foundBadValue: goodIndices.append(i)
	goodIndices = list(set(goodIndices))
	goodIndices.sort()
	#print "goodIndices", goodIndices
	
	for j in xrange(len(arrayList)):
		arrayList[j] = arrayList[j][goodIndices]
	
	return arrayList

def valuesEq(x,y):
	for i in xrange(len(x)):
		for j in xrange(len(x[0])):
			if x[i][j] != y[i][j]: return False
	return True
		
def testClean():
	a = np.array([1,2,5,4,-9999,4,1,2])
	b = np.array([1,2,9,4,-9999,4,-9999,2])
	c = np.array([2,-9999,5,4,-2,8,1,2])
	d = np.array([1,5,5,4,-2,4,1,6])
	#print "clean([a,b,c,d])",  clean([a,b,c,d])
	assert valuesEq(clean([a,b,c,d]), [[1,5,4,4,2], [1,9,4,4,2], [2,5,4,8,2], [1,5,4,4,6]])
		

if __name__ == "__main__":
	df = datafile("/Users/janagrc/desktop/ssppOut-dr9.fits")
	print "Spectral types: ", df.uniqueSpectralTypes()
	df.plotHist()

spect3.py
from astropy.io import fits
import numpy as np
import copy
import pylab
import math

class Star(object):
	def __init__(self):
		self.fields = {}
	
#class DataFile(object,requiredfields=None):
class DataFile(object):
	def __init__(self, path):
		self.path = path
		datafile = fits.open(path)
		column_names = datafile[1].columns.names
		self.stars = [] #list of star objects, each with a dict of fields
		#for row in xrange(0,len(datafile[1].data.field('TEFF_ADOP')+1):
		x = True
		row=0
		print x
		#while x is True:
		#    try:
		#        stardata=datafile[1].data[row]
		#        if goodStar(stardata):
		        #    s=Star()
		        #    s.fields[column]=stardata.field(column)
		        #    self.stars.append(s)
		        #    row=row+1
		#    except IndexError:
		#        x=False
		        
def goodStar(stardata, badValue=-9999):
    goodvalues=True
    if (stardata["LOGG_ADOP"] == badValue) or (stardata["FEH_ADOP"] == badValue) or (stardata["TEFF_ADOP"]==badvalue):
        goodvalues=False
    return goodvalues
    
    
#### Good to here    
    
if __name__ == "__main__":
	df = DataFile("../data/ssppOut-dr9.fits")

spect.py
from astropy.io import fits
import numpy as np
import copy
import pylab
import math


class DataFile:
	def __init__(self, path, variableList=["LOGG_ADOP", "FEH_ADOP", "TEFF_ADOP", "SPECTYPE_HAMMER"]):
	#Input a check for the path?
		self.path = path
		datafile = fits.open(path)
		#datafile.fields
		self.logg = 
		
		
		
		
		np.array(datafile[1].data.field("LOGG_ADOP"))
		self.feh = np.array(datafile[1].data.field("FEH_ADOP"))
		self.teff = np.array(datafile[1].data.field("TEFF_ADOP"))
		self.spectypehammer = np.array(datafile[1].data.field("SPECTYPE_HAMMER"))
		self.clean()
		
	def clean(self):
		self.logg, self.feh, self.teff, self.spectypehammer = cleandata([self.logg, self.feh, self.teff, self.spectypehammer])
		
	def unique_spectral_types(self):
		return list(set(self.spectypehammer))
		
	def plothist(self):
		fig,axes = pylab.subplots(1, 3)
		axes[0].hist(self.feh, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled', bins=20)
		axes[1].hist(self.feh, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled')
		axes[2].hist(self.feh, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled')
		pylab.show()
	

def cleandata(arrayList, badValue=-9999):
	goodIndices = []
	for i in xrange(len(arrayList[0])):
		foundBadValue = False
		for array in arrayList:
			assert len(arrayList[0]) == len(array)
			if array[i] == badValue: foundBadValue = True
		if not foundBadValue: goodIndices.append(i)
	goodIndices = list(set(goodIndices))
	goodIndices.sort()
	#print "goodIndices", goodIndices
	
	for j in xrange(len(arrayList)):
		arrayList[j] = arrayList[j][goodIndices]
	
	return arrayList

def values_eq(x,y):
	for i in xrange(len(x)):
		for j in xrange(len(x[0])):
			if x[i][j] != y[i][j]: return False
	return True
		
		

if __name__ == "__main__":
	df = DataFile("/Users/janagrc/desktop/ssppOut-dr9.fits")
	print "Spectral types: ", df.unique_spectral_types()
	df.plothist()
	
	
	
	
# STUFF FROM THE END OF SPECT3 
	
	
	   
        if 
    if  stardata[column
			if array[i] == badValue: foundBadValue = True
		if not foundBadValue: goodIndices.append(i)
	goodIndices = list(set(goodIndices))
	goodIndices.sort()
	#print "goodIndices", goodIndices
	
	for j in xrange(len(arrayList)):
		arrayList[j] = arrayList[j][goodIndices]
	
	return arrayList
			
			
			for column in column_names:
				s.fields[column]=stardata.field(column)
			    if isgood(s).flag:
				    self.stars.append(s)
		# for row in rows
		# read in row
		# check it's good if isgood(star) then 
		
		# if good, add star object to stars
			
	def isgood(self):
		self.flag= True
		for field in requiredFields:
		    if self.field == -9999:
		        self.flag = False
		
		#Fields X y and z cannot be -9999
		#if anything is bad set to false
	
	def clean(self):
		self.logg, self.feh, self.teff, self.spectypehammer = cleanData([self.logg, self.feh, self.teff, self.spectypehammer])
		#cleanedDataList = cleanData(self.dataDict.values())			
		self.logg, self.feh, self.teff, self.spectypehammer = cleandata([self.logg, self.feh, self.teff, self.spectypehammer])
		
	def unique_spectral_types(self):
		return list(set(self.spectypehammer))
		
	def plothist(self):
		fig,axes = pylab.subplots(1, 3)
		axes[0].hist(self.feh, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled', bins=20)
		axes[1].hist(self.feh, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled')
		axes[2].hist(self.feh, label="[Fe/H]", facecolor="red", alpha=0.5, histtype='stepfilled')
		pylab.show()
	


def values_eq(x,y):
	for i in xrange(len(x)):
		for j in xrange(len(x[0])):
			if x[i][j] != y[i][j]: return False
	return True
		
def test_clean():
	a = np.array([1,2,5,4,-9999,4,1,2])
	b = np.array([1,2,9,4,-9999,4,-9999,2])
	c = np.array([2,-9999,5,4,-2,8,1,2])
	d = np.array([1,5,5,4,-2,4,1,6])
	#print "clean([a,b,c,d])",  clean([a,b,c,d])
	assert values_eq(clean([a,b,c,d]), [[1,5,4,4,2], [1,9,4,4,2], [2,5,4,8,2], [1,5,4,4,6]])
		

if __name__ == "__main__":
	df = DataFile("/Users/janagrc/desktop/ssppOut-dr9.fits")
	print "Spectral types: ", df.unique_spectral_types()
	df.plothist()
